inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -71,7 +74,15 @@ def arxiv_fulltext_download(obj, eng):
     )
 
     if pdf:
-        obj.log.info('PDF retrieved from arXiv for %s', arxiv_id)
+        if pdf.get_version().mimetype == 'application/pdf':
+            obj.log.info('PDF retrieved from arXiv for %s', arxiv_id)
+        else:
+            # We need to delete the failed PDF
+            del obj.files[filename]
+            pdf.delete()
+            warning_message = "PDF from arXiv for %s isn't yet ready"
+            obj.log.warning(warning_message, arxiv_id)
+            raise RuntimeError(warning_message % arxiv_id)
```

Comment:
I think it's time to introduce the `WorkflowsTaskError` I was thinking of introducing in my last PR. It's certainly more expressive than a RuntimeError. BTW, does `backoff` respect inheritance?